### PR TITLE
Improve parameter handling in UsersManager.getUsers

### LINKED ISSUE #13507: Error message in various places when invalid parameters are used
as seen on production many times:

## first error message (more below)
Argument 1 passed to Piwik\\Plugins\\UsersManager\\Model::getUsers() must be of the type array, null given, called in /plugins\/UsersManager\/API.php on line 351

URL: index.php?module=API&method=UsersManager.getUsers&format=Tsv&token_auth=XYZANONYMIZED&translateColumnNames=1&userLogins[]=&userLogins[]=

### PR DESCRIPTION
**Pull request by sgiehl**
aims to fix https://github.com/matomo-org/matomo/issues/13507#issue-365156020

### CODE REVIEW

**Comment by diosmosis** [comment]
Can a test be added to UsersManager/tests/Integration/APITest.php?

File: plugins/UsersManager/API.php [hunk 1]
```diff
@@ -336,15 +336,18 @@ public function getUsersPlusRole($idSite, $limit = null, $offset = 0, $filter_se
...
      * @return array the list of all the users
      */
     public function getUsers($userLogins = '')
     {
         Piwik::checkUserHasSomeAdminAccess();
 
         $logins = array();
-        if (!empty($userLogins)) {
+
+        if (is_array($userLogins)) {
```
Comment (tsteur) [inline plugins/UsersManager/API.php]: or should we just fail if userLogins is not a string? In theory you may be able to send now some weird array structures to maybe cause an error? (haven't tested, just a thought)

Comment (sgiehl) [reply plugins/UsersManager/API.php]: don't have a strong opinion on that. failing if no string is given works as well for sure

Comment (tsteur) [reply plugins/UsersManager/API.php]: lets do this to keep things simple 👍

**Review by tsteur — APPROVED** [review]

**Pull request merged**